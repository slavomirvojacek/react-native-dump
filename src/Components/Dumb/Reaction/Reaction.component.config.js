import PropTypes from 'prop-types'

// import { ReactionType } from '../../Constants/ReactionType'
// import { issueDef } from '../../../defs'
// import { noop } from '../../../../../Utils/misc'

const displayName = 'ReactionComponent'

const propTypes = {
  type: PropTypes.string.isRequired,
  addReactionToIssue: PropTypes.func.isRequired,
  removeReactionFromIssue: PropTypes.func.isRequired,
  pressed: PropTypes.bool.isRequired,
  issue: PropTypes.object.isRequired,
}

// Only used for testing
// const testDefaultProps = {
//   type: ReactionType.DEFAULT,
//   addReactionToIssue: noop,
//   removeReactionFromIssue: noop,
//   pressed: false,
//   issue: issueDef(),
// }

export default {
  displayName, propTypes
}
