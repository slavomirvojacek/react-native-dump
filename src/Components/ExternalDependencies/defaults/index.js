import { noop } from '@openmaths/utils'

const props = {
  logIntoConsole: noop,
}

export {
  props
}
