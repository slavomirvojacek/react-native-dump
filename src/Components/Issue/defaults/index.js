const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: {
      nodes: []
    }
  }

  return {...def, ...override}
}

const props = {
  issue: issueDefault(),
}

export {
  issueDefault, props
}
