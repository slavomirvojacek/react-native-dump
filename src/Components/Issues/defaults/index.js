const props = {
  issues: [],
}

export {
  props
}
