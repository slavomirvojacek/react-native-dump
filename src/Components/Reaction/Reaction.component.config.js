import PropTypes from 'prop-types'

// import { ReactionType } from '../../Constants/ReactionType'
// import { issueDefault } from '../../../defs'
// import { noop } from '../../../../../Utils/misc'

const displayName = 'ReactionComponent'

const propTypes = {
  type: PropTypes.string.isRequired,
  addReactionToIssue: PropTypes.func.isRequired,
  removeReactionFromIssue: PropTypes.func.isRequired,
  pressed: PropTypes.bool.isRequired,
  issue: PropTypes.object.isRequired,
}

// Only used for testing
// const testDefaultProps = {
//   type: ReactionType.DEFAULT,
//   addReactionToIssue: noop,
//   removeReactionFromIssue: noop,
//   pressed: false,
//   issue: issueDefault(),
// }

export default {
  displayName, propTypes
}
