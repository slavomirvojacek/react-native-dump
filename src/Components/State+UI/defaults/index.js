const props = {
  color: 'black', // has to be a valid colour, otherwise warnings are thrown
  isWarningVisible: false,
}

const state = {
  ...props,
  color: 'green', // has to be a valid colour, otherwise warnings are thrown
}


export {
  props, state
}
