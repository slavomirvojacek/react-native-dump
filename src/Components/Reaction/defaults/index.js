import { noop } from '@openmaths/utils'

import { issueDefault } from '../../Issue/defaults'
import { ReactionType } from '../../../Constants/ReactionType'

const userDefault = (override = {}) => {
  const def = {
    id: '',
  }

  return {...def, ...override}
}

const reactionDefault = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDefault()
  }

  return {...def, ...override}
}

const props = {
  type: ReactionType.DEFAULT,
  addReactionToIssue: noop,
  removeReactionFromIssue: noop,
  pressed: false,
  issue: issueDefault()
}

export {
  userDefault, reactionDefault, props
}
