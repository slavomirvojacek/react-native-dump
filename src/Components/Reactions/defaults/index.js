import { None } from '@threestup/monads'

import { issueDefault } from '../../Issue/defaults'

const props = {
  issue: issueDefault(),
  me: None
}

export {
  props
}
