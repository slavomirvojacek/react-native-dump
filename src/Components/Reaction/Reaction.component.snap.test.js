// import { Some } from '@threestup/monads'

import Reaction from './Reaction.component'

import { props as testDefaultProps } from './defaults'
import { ReactionType } from '../../Constants/ReactionType'
import { assertSnapshots } from '../../TestUtils/snapshot'

describe('Reaction Snapshots', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const configs = [
    {
      props: {...testDefaultProps},
      desc: 'throws if type is not one of `THUMBS_UP`, `THUMBS_DOWN`',
      throws: TypeError
    },
    {
      props: {
        ...testDefaultProps,
        type: ReactionType.THUMBS_UP
      },
      desc: 'renders correctly when type is `THUMBS_UP`',
    },
    {
      props: {
        ...testDefaultProps,
        type: ReactionType.THUMBS_DOWN
      },
      desc: 'renders correctly when type is `THUMBS_DOWN`',
    },
    {
      props: {
        ...testDefaultProps,
        type: ReactionType.THUMBS_UP,
        pressed: true
      },
      desc: 'renders correctly when type is `THUMBS_UP` and pressed is `true`',
    },
    {
      props: {
        ...testDefaultProps,
        type: ReactionType.THUMBS_DOWN,
        pressed: true
      },
      desc: 'renders correctly when type is `THUMBS_DOWN` and pressed is `true`',
    },
  ]

  assertSnapshots(Reaction, configs)
})
