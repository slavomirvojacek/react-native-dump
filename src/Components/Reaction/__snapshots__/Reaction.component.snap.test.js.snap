// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reaction Snapshots renders correctly when type is \`THUMBS_DOWN\` 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "fontSize": 14,
      "height": 40,
      "marginRight": 5,
      "width": 40,
    }
  }
>
  👎
</Text>
`;

exports[`Reaction Snapshots renders correctly when type is \`THUMBS_DOWN\` and pressed is \`true\` 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "blue",
      "fontSize": 14,
      "height": 40,
      "marginRight": 5,
      "width": 40,
    }
  }
>
  👎
</Text>
`;

exports[`Reaction Snapshots renders correctly when type is \`THUMBS_UP\` 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "fontSize": 14,
      "height": 40,
      "marginRight": 5,
      "width": 40,
    }
  }
>
  👍
</Text>
`;

exports[`Reaction Snapshots renders correctly when type is \`THUMBS_UP\` and pressed is \`true\` 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "blue",
      "fontSize": 14,
      "height": 40,
      "marginRight": 5,
      "width": 40,
    }
  }
>
  👍
</Text>
`;
